### PR TITLE
Offline notification

### PR DESCRIPTION
**Pull request by FlorianBouron**

### CODE REVIEW

File: packages/extension/src/components/molecules/NetworkIndicator/NetworkIndicator.test.tsx [hunk 1]
```diff
@@ -27,20 +28,22 @@ describe('NetworkIndicator', () => {
...
-      switchNetwork: jest.fn()
+      networkName: 'Loading...',
+      switchNetwork: jest.fn(),
+      isConnectionFailed: false
     });
 
     render(<NetworkIndicator />);
 
-    expect(screen.getByText('Switch network')).toBeInTheDocument();
+    expect(screen.getByText('Loading...')).toBeInTheDocument();
```
Comment (ThibautBremand) [inline packages/extension/src/components/molecules/NetworkIndicator/NetworkIndicator.test.tsx:38]: 👍

File: packages/extension/src/components/pages/AddNewWallet/ConfirmPassword/ConfirmPassword.tsx [hunk 2]
```diff
@@ -47,7 +48,7 @@ export const ConfirmPassword: FC<ConfirmPasswordProps> = ({ setPassword, onConfi
           display: 'flex',
           flexDirection: 'column',
           justifyContent: 'space-between',
-          height: '100%',
+          height: isConnectionFailed ? 'calc(100% - 56px)' : '100%',
```
Comment (ThibautBremand) [inline packages/extension/src/components/pages/AddNewWallet/ConfirmPassword/ConfirmPassword.tsx:51]: Maybe 56px could be extracted as a const somewhere

Comment (FlorianBouron) [reply packages/extension/src/components/pages/AddNewWallet/ConfirmPassword/ConfirmPassword.tsx:51]: Good point! I will do this tomorrow!

File: packages/extension/src/components/templates/AsyncTransaction/AsyncTransaction.tsx [hunk 3]
```diff
@@ -56,11 +58,12 @@ export const AsyncTransaction: FC<AsyncTransactionProps> = ({
     <Container
       component="main"
       style={{
+        ...(isConnectionFailed ? { position: 'fixed', top: 56 } : {}),
         display: 'flex',
         flexDirection: 'column',
         justifyContent: 'space-between',
         alignItems: 'center',
-        height: '100vh',
+        height: isConnectionFailed ? `calc(100vh - 56px)` : '100vh',
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/AsyncTransaction/AsyncTransaction.tsx:66]: Same here for 56px that could be extracted somewhere

File: packages/extension/src/components/templates/ErrorBoundary/ErrorBoundary.tsx [hunk 4]
```diff
@@ -42,8 +45,14 @@ export class ErrorBoundary extends Component<ErrorBoundaryProps, State> {
             display: 'flex',
             flexDirection: 'column',
             justifyContent: 'space-between',
-            height: '100vh',
-            padding: '15x 16px'
+            padding: '15x 16px',
+            ...(isConnectionFailed
+              ? {
+                  height: 'calc(100vh - 56px)',
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/ErrorBoundary/ErrorBoundary.tsx:51]: Same here with 56px

File: packages/extension/src/components/templates/PageWithHeader/PageWithHeader.tsx [hunk 5]
```diff
@@ -16,19 +16,30 @@ export interface PageWithHeaderProps {
...
   }
   return (
-    <>
+    <div
+      style={
+        isConnectionFailed
+          ? {
+              width: '100%',
+              position: 'fixed',
+              top: 56
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithHeader/PageWithHeader.tsx:31]: 56px

File: packages/extension/src/components/templates/PageWithHeader/PageWithHeader.tsx [hunk 6]
```diff
@@ -16,19 +16,30 @@ export interface PageWithHeaderProps {
...
+      }
+    >
       <Header wallet={wallets[selectedWallet]} />
       <Container
         component="main"
         style={{
           display: 'flex',
           flexDirection: 'column',
-          height: `calc(100vh - ${CONTAINER_HEIGHT_TAKEN}px)`,
+          height: `calc(100vh - ${CONTAINER_HEIGHT_TAKEN}px${isConnectionFailed ? ' - 56px' : ''})`,
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithHeader/PageWithHeader.tsx:42]: 56px

File: packages/extension/src/components/templates/PageWithNavMenu/PageWithNavMenu.tsx [hunk 7]
```diff
@@ -1,19 +1,23 @@
...
 
 export const PageWithNavMenu: FC<PageWithNavMenuProps> = ({ children, indexDefaultNav }) => {
+  const { isConnectionFailed } = useNetwork();
+
   return (
     <>
       <div
         style={{
-          height: `calc(100% - ${NAV_MENU_HEIGHT}px)`,
+          height: `calc(100% - ${NAV_MENU_HEIGHT}px${isConnectionFailed ? ' - 56px' : ''})`,
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithNavMenu/PageWithNavMenu.tsx:18]: 56px

File: packages/extension/src/components/templates/PageWithNavMenu/PageWithNavMenu.tsx [hunk 8]
```diff
@@ -1,19 +1,23 @@
...
           position: 'fixed',
+          top: isConnectionFailed ? 56 : undefined,
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithNavMenu/PageWithNavMenu.tsx:20]: 56px

File: packages/extension/src/components/templates/PageWithReturn/PageWithReturn.tsx [hunk 9]
```diff
@@ -21,8 +23,19 @@ export const PageWithReturn: FC<PageWithReturnProps> = ({
...
+  const { isConnectionFailed } = useNetwork();
+
   return (
-    <>
+    <div
+      style={
+        isConnectionFailed
+          ? {
+              position: 'fixed',
+              top: 56
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithReturn/PageWithReturn.tsx:34]: 56px

File: packages/extension/src/components/templates/PageWithStepper/PageWithStepper.tsx [hunk 10]
```diff
@@ -38,7 +40,17 @@ export const PageWithStepper: FC<PageWithStepperProps> = ({
...
 
   return (
-    <>
+    <div
+      style={
+        isConnectionFailed
+          ? {
+              width: '100%',
+              position: 'fixed',
+              top: 56
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithStepper/PageWithStepper.tsx:49]: 56px

File: packages/extension/src/components/templates/PageWithStepper/PageWithStepper.tsx [hunk 11]
```diff
@@ -51,15 +63,15 @@ export const PageWithStepper: FC<PageWithStepperProps> = ({
             Back
           </Button>
         }
-        style={{ backgroundColor: '#282c34' }}
+        style={{ backgroundColor: '#282c34', ...(isConnectionFailed ? { top: 56 } : {}) }}
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithStepper/PageWithStepper.tsx:66]: 56px

File: packages/extension/src/components/templates/PageWithStepper/PageWithStepper.tsx [hunk 12]
```diff
@@ -51,15 +63,15 @@ export const PageWithStepper: FC<PageWithStepperProps> = ({
...
       />
       <Container
         component="main"
         style={{
           display: 'flex',
           flexDirection: 'column',
           justifyContent: 'space-between',
-          height: '100vh',
+          height: isConnectionFailed ? `calc(100vh - 56px)` : '100vh',
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithStepper/PageWithStepper.tsx:74]: 56px

File: packages/extension/src/components/templates/PageWithTitle/PageWithTitle.tsx [hunk 13]
```diff
@@ -10,14 +12,17 @@ export interface PageWithTitleProps {
 }
 
 export const PageWithTitle: FC<PageWithTitleProps> = ({ title, children, styles }) => {
+  const { isConnectionFailed } = useNetwork();
+
   return (
     <Container
       component="main"
       style={{
+        ...(isConnectionFailed ? { position: 'fixed', top: 56 } : {}),
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithTitle/PageWithTitle.tsx:21]: 56px

File: packages/extension/src/components/templates/PageWithTitle/PageWithTitle.tsx [hunk 14]
```diff
@@ -10,14 +12,17 @@ export interface PageWithTitleProps {
...
         display: 'flex',
         flexDirection: 'column',
         justifyContent: 'space-between',
-        height: '100vh',
+        height: isConnectionFailed ? 'calc(100vh - 56px)' : '100vh',
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/PageWithTitle/PageWithTitle.tsx:25]: 56px

File: packages/extension/src/components/templates/AsyncTransaction/AsyncTransaction.tsx [hunk 15]
```diff
@@ -56,11 +58,12 @@ export const AsyncTransaction: FC<AsyncTransactionProps> = ({
     <Container
       component="main"
       style={{
+        ...(isConnectionFailed ? { position: 'fixed', top: 56 } : {}),
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/AsyncTransaction/AsyncTransaction.tsx:61]: 56px

File: packages/extension/src/components/templates/ErrorBoundary/ErrorBoundary.tsx [hunk 16]
```diff
@@ -42,8 +45,14 @@ export class ErrorBoundary extends Component<ErrorBoundaryProps, State> {
...
+                  position: 'fixed',
+                  top: 56
```
Comment (ThibautBremand) [inline packages/extension/src/components/templates/ErrorBoundary/ErrorBoundary.tsx:53]: 56px

**Review by ThibautBremand — APPROVED** [review]

**Pull request merged**